lib/config/config-file.js

```diff
@@ -472,13 +498,14 @@ function resolve(filePath, relativeTo) {
     let normalizedPackageName;
 
     if (filePath.indexOf("plugin:") === 0) {
+        const pluginPath = filePath;
         const packagePath = filePath.substr(7, filePath.lastIndexOf("/") - 7);
         const configName = filePath.substr(filePath.lastIndexOf("/") + 1, filePath.length - filePath.lastIndexOf("/") - 1);
 
         normalizedPackageName = normalizePackageName(packagePath, "eslint-plugin");
         debug(`Attempting to resolve ${normalizedPackageName}`);
         filePath = resolver.resolve(normalizedPackageName, getLookupPath(relativeTo));
-        return { filePath, configName };
+        return { filePath, configName, pluginPath };
```

Comment:
I'm not sure if there's an official JSDoc way to do it, but just explaining the optional properties in words would be a good improvement.